### PR TITLE
Version 2.0

### PR DESCRIPTION
**Pull request by cullophid**
New parse function.
Changed code style to match js standard
changed all functions returning errors to return invalid dates of NaN as appropriate

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@ node_modules/
 dist/
 build/
 sandbox.js
+.DS_Store
```
Comment (davidchambers) [inline .gitignore:5]: This should really live in your [global **.gitignore**](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore).

Comment (cullophid) [inline .gitignore:5]: Thats a valid point

**Comment by TheLudd** [comment]
Is this introducing https://github.com/feross/standard? Because I thought the point was that you should not need an eslint configuration file.

**Comment by cullophid** [comment]
@TheLudd Its introducing the standard of js standard :)
The only real change is the lack of semi colons. I added it in because i like the style, and wanted to hear what people thought.

I realize that This PR is way too big. I am probably gonna split it up but i wanted to hear what people though first.

**Comment by TheLudd** [comment]
Splitting up is a good idea. 

I like no semi colons as long as there is an automatic checker for the cases where no semicolons cause trouble.

**Comment by cullophid** [comment]
We have: no-unexpected-multiline: 2  # - Avoid code that looks like two expressions but is actually one
:)

**Comment by cullophid** [comment]
Im gonna run this as 3 separate PR instead

**Pull request closed without merging**